Makefile

```diff
@@ -0,0 +1,27 @@
+RESET_TARGET := upstream/master
+PKGS := $(shell go list ./... | grep -v /vendor/)
+override BRANCH := $(shell sh -c "git branch | sed -n 's|* \(.*\)|\1|p'")
+
+.PHONY: squash format test
+
+help:
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
+
+ifeq ($(COMMIT),)
+squash:
+	@echo "COMMIT=<COMMIT> not provided. Exiting..."
+	@exit 1
+else
+
+# Assuming PR is ready, everything added as intended, no conflicts
+squash: test format ## Runs test, format, and force pushes. Override RESET_TARGET='upstream/master' if necessary
+	git reset --soft $(RESET_TARGET)
+	git commit -m "$(COMMIT)"
+	git push --force origin $(BRANCH)
+endif
+
+format: ## Runs go fmt on all non-vendor go packages
+	go fmt $(PKGS)
+
+test: ## Runs `go test -count=1 -race -v ./pkg/...`
+	go test -count=1 -race -v ./pkg/...
```

Comment:
This Makefile look fine although will want to be documented (e.g. when do folks use it?), and how about an extra target for running the Docker tests (now we have Dockerfile_test and Dockerfile_race_test)?

Reply:
I think it's best suited to DEVELOPMENT.md

I suggest `make test` runs our main Dockerfile_test and if you do `make race test` it runs both our main Dockerfile_test and the Dockerfile_race_test (so they're both independent targets, but you can combine them if you want)

A "final check before we merge" would involve `make race test` for example, so we get all the coverage. Lemme know what you think.

Reply:
Both of the dockerized files have nearly the same execution time
`time docker build -f Dockerfile_race_test .`=`real	2m2.667s`
`time docker build -f Dockerfile_test .`=`real	2m3.154s`
A final check (one that handles race conditions) makes sense, but is there a particular reason we don't consolidate to one of these files?
I always prefer to keep things as lean as possible, same regards for the Makefile. We could have `make test`, which would be undockerized and `make docker/test` will run our racefile? I made some changes.

Reply:
`A final check (one that handles race conditions) makes sense, but is there a particular reason we don't consolidate to one of these files?`

Can't run the race tests in the Alpine image

Yeah I like `make test` undockerized and `make docker/test` as running *both* the Dockerfiles - we don't want to lose test coverage (one is Alpine which we "ship on" but w/o race checking, one is Stretch which we don't "ship on" but lets us do the race checks)

I say `ship on` in quotes as we're gonna end up releasing images eventually and they're gonna be Alpine for running on our target platforms I bet